rosidl_typesupport_introspection_tests/test/introspection_libraries_under_test.hpp

```diff
@@ -45,12 +48,29 @@
 #include "rosidl_typesupport_introspection_tests/msg/multi_nested.hpp"
 #include "rosidl_typesupport_introspection_tests/msg/strings.hpp"
 #include "rosidl_typesupport_introspection_tests/msg/unbounded_sequences.hpp"
+#include "rosidl_typesupport_introspection_tests/srv/arrays.hpp"
+#include "rosidl_typesupport_introspection_tests/srv/basic_types.hpp"
+#include "rosidl_typesupport_introspection_tests/srv/empty.hpp"
 
 #include "rosidl_typesupport_introspection_tests/fixtures.hpp"
 #include "rosidl_typesupport_introspection_tests/helpers.hpp"
 #include "rosidl_typesupport_introspection_tests/libraries.hpp"
 #include "rosidl_typesupport_introspection_tests/type_traits.hpp"
 
+// Extra C++ APIs to homogeneize access to test interfaces in C and C++
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, Arrays)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, BasicTypes)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, BoundedSequences)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, Constants)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, Defaults)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, Empty)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, MultiNested)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, Strings)
+DEFINE_CXX_API_FOR_C_MESSAGE(rosidl_typesupport_introspection_tests, msg, UnboundedSequences)
+DEFINE_CXX_API_FOR_C_SERVICE(rosidl_typesupport_introspection_tests, srv, Arrays)
+DEFINE_CXX_API_FOR_C_SERVICE(rosidl_typesupport_introspection_tests, srv, BasicTypes)
+DEFINE_CXX_API_FOR_C_SERVICE(rosidl_typesupport_introspection_tests, srv, Empty)
+
```

Comment:
Out of curiosity, what is the reason not to include the same types in `srv` than the ones used in `msg`?  Only Arrays, BasicTypes and Empty.

Reply:
@j-rivero only these exist. See https://github.com/ros2/test_interface_files/tree/master/srv.